gluten-flink/planner/src/main/java/org/apache/gluten/rexnode/Utils.java

```diff
@@ -46,4 +56,42 @@ public static void registerRegistry() {
             ISerializableRegistry.registerAll();
         }
     }
+
+    private static final List<Type> NUMBER_TYPE_PRIORITY_LIST = List.of(
+        new TinyIntType(),
+        new SmallIntType(),
+        new IntegerType(),
+        new BigIntType(),
+        new RealType(),
+        new DoubleType()
+    );
+
+    private static int getNumberTypePriority(Type type) {
+        int index = -1;
+        for (int i = 0; i < NUMBER_TYPE_PRIORITY_LIST.size(); ++i) {
+            if (NUMBER_TYPE_PRIORITY_LIST.get(i).getClass().equals((type.getClass()))) {
+                index = i;
+                break;
+            }
+        }
+        if (index == -1) {
+            throw new RuntimeException("Unsupported type: " + type.getClass().getName());
+        }
+        return index;
+    }
+
+    public static List<TypedExpr> promoteTypeForNumberExpressions(List<TypedExpr> expressions) {
```

Comment:
Assume this is only for arithmetic expressions. So suggest to name this method like:
promoteTypeForArithmeticExpressions

Reply:
done